homeassistant/components/knx/websocket.py

```diff
@@ -541,3 +546,115 @@ def ws_create_device(
         configuration_url=f"homeassistant://knx/entities/view?device_id={_device.id}",
     )
     connection.send_result(msg["id"], _device.dict_repr)
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "knx/get_entity_schemas",
+    }
+)
+@provide_knx
+@callback
+def ws_get_entity_schemas(
+    hass: HomeAssistant,
+    knx: KNXModule,
+    connection: websocket_api.ActiveConnection,
+    msg: dict,
+) -> None:
+    """Retrieve schema definitions for generating dynamic forms in the UI.
+
+    This WebSocket command provides schema definitions required for configuring
+    KNX entities in Home Assistant. The schemas are designed to enable the frontend
+    to dynamically generate forms for user input, ensuring that the configuration
+    matches the expected structure and validation rules.
+
+    The schemas are retrieved dynamically using the `get_schema` method from
+    configuration classes (e.g., `UiSensorConfig`). They are serialized into a
+    JSON-compatible format via the `SchemaSerializer.convert` method and sent to
+    the frontend through the WebSocket connection.
+
+    Args:
+        hass: The Home Assistant core object.
+        knx: The KNX integration module, provided via the
+             `@provide_knx` decorator.
+        connection: The WebSocket connection
+                    instance to communicate with the frontend.
+        msg: The incoming WebSocket message containing the command details.
+
+    Raises:
+        ValueError: If there is an issue serializing the schemas.
+
+    """
+
+    supportedConfigs: tuple[type[BasePlatformConfiguration], ...] = (UiSensorConfig,)
+
+    payload = tuple(
+        SchemaSerializer.convert(config.get_schema()) for config in supportedConfigs
+    )
+
+    connection.send_result(msg["id"], payload)
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "knx/create_entity2",
+        vol.Required(CONF_DATA): vol.Schema(vol.Any(UiSensorConfig.get_schema())),
```

Comment:
```suggestion
        vol.Required(CONF_DATA): dict,
```
The validation is happening in the function, no need to validate twice - and we couldn't do the custom exception handling if this didn't validate here.

Reply:
Alright, that makes life easier. Should we keep a separate endpoint for the new logic, or should I merge the create endpoint with the current one?

Reply:
Having different endpoints sounds like a good idea until the other platforms (switch, light, maybe binary_sensor) are migrated to the new system - which should be separate PRs.